src/main/java/com/alkemy/ong/security/MainUser.java

```diff
@@ -0,0 +1,99 @@
+package com.alkemy.ong.security;
+
+import com.alkemy.ong.domain.Role;
+import com.alkemy.ong.repository.model.UserModel;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MainUser  implements UserDetails {
+
+    private String firstName;
+    private String lastName;
+    private String email;
+    private String password;
+    private String photo;
+    private Collection<?extends GrantedAuthority> authorities;
+    private LocalDateTime creationDate;
+
+    public static MainUser build( UserModel userModel) {
+        List<GrantedAuthority> authorities =
+                userModel.getRole().stream().map(role -> {
+                    return new
+                            SimpleGrantedAuthority((role.getRoleName()).name());
+                }).collect(Collectors.toList());
```

Comment:
Estas queriendo hacer un `.stream().map` de un atributo que es solo un RoleModel. El `stream().map` se aplica sobre collecciones, por eso no estaba funcionando 👌🏼  Con esto debería funcionar:
```suggestion
        List<GrantedAuthority> authorities = List.of(new SimpleGrantedAuthority((userModel.getRole().getRoleName()).name()));
```